code/modules/shuttle/shuttle.dm

```diff
@@ -849,20 +845,21 @@
 	admin_controlled = 1
 	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
 
-/obj/machinery/computer/shuttle/ferry/request/Topic(href, href_list)
+/obj/machinery/computer/shuttle/ferry/request/tgui_act(action, list/params, datum/tgui/ui, datum/tgui_state/state)
 	if(..())
-		return 1
-	if(href_list["request"])
+		return
+	if(action == "request")
 		if(cooldown)
 			return
-		cooldown = 1
+		cooldown = TRUE
 		to_chat(usr, "<span class='notice'>Your request has been recieved by Centcom.</span>")
 		log_admin("[key_name(usr)] requested to move the transport ferry to Centcom.")
 		message_admins("<b>FERRY: <font color='blue'>[key_name_admin(usr)] (<A HREF='?_src_=holder;secretsfun=moveferry'>Move Ferry</a>)</b> is requesting to move the transport ferry to Centcom.</font>")
-		. = 1
-		SSnanoui.update_uis(src)
+		. = TRUE
+		SStgui.update_uis(src)
 		spawn(600) //One minute cooldown
```

Comment:
Using `addtimer` is preferable here. You could even get rid of the `cooldown` variable and instead use `world.timer` and adding whatever duration you want. Do make a define for the cooldown, avoid magic numbers.

Also `add_fingerprint(usr)`